pkg/external_tools/mcp/pwru_test.go

```diff
@@ -0,0 +1,281 @@
+			filter:           "tcp port $(echo 80)",
+			outputLimitLines: 100,
+			wantError:        true,
+		},
+		{
+			name:             "invalid filter - newline injection",
+			filter:           "tcp port 80\nrm -rf /",
+			outputLimitLines: 100,
+			wantError:        true,
+		},
+		{
+			name:             "invalid filter - null byte injection",
+			filter:           "tcp port 80\x00rm -rf /",
+			outputLimitLines: 100,
+			wantError:        true,
+		},
+		// Output limit lines validation tests
+		{
+			name:             "zero output limit lines - should use default",
+			filter:           "",
+			outputLimitLines: 0,
+			wantError:        false,
+		},
+		{
+			name:             "negative output limit lines",
+			filter:           "",
+			outputLimitLines: -1,
+			wantError:        true,
+		},
+		{
+			name:             "minimum valid output limit lines",
+			filter:           "",
+			outputLimitLines: 1,
+			wantError:        false,
+		},
+		{
+			name:             "output limit lines within limit",
+			filter:           "",
+			outputLimitLines: 500,
+			wantError:        false,
+		},
+		{
+			name:             "output limit lines at max",
+			filter:           "",
+			outputLimitLines: MaxOutputLimitLines,
+			wantError:        false,
+		},
+		{
+			name:             "output limit lines just over max",
+			filter:           "",
+			outputLimitLines: MaxOutputLimitLines + 1,
+			wantError:        true,
+		},
+		{
+			name:             "output limit lines exceeds max",
+			filter:           "",
+			outputLimitLines: 1500,
+			wantError:        true,
+		},
+		{
+			name:             "very large output limit lines",
+			filter:           "",
+			outputLimitLines: 999999,
+			wantError:        true,
+		},
+		// Combined validation tests
+		{
+			name:             "valid filter and valid output limit",
+			filter:           "tcp port 80",
+			outputLimitLines: 500,
+			wantError:        false,
+		},
+		{
+			name:             "valid filter but invalid output limit",
+			filter:           "tcp port 80",
+			outputLimitLines: 2000,
+			wantError:        true,
+		},
+		{
+			name:             "invalid filter but valid output limit",
+			filter:           "tcp port 80; ls",
+			outputLimitLines: 500,
+			wantError:        true,
+		},
+		{
+			name:             "invalid filter and invalid output limit",
+			filter:           "tcp port 80; ls",
+			outputLimitLines: 2000,
+			wantError:        true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Validate filter
+			err := ValidateBPFFilter(tt.filter)
+			if err != nil {
+				if !tt.wantError {
+					t.Errorf("ValidateBPFFilter(%q) error = %v, wantError %v", tt.filter, err, tt.wantError)
+				}
+				return
+			}
+
+			// Validate output limit lines
+			outputLimitLines := tt.outputLimitLines
+			if outputLimitLines == 0 {
+				outputLimitLines = MaxOutputLimitLines
+			}
+			err = validateIntMax(outputLimitLines, MaxOutputLimitLines, "output_limit_lines", "")
```

Comment:
Since this function is already tested in `utils_test.go`, let's remove the test from here.